src/Microsoft.Health.Fhir.SqlServer/Features/Schema/ResourceVersion.cs

```diff
@@ -0,0 +1,32 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using Microsoft.Data.SqlClient;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Schema
+{
+    public static class ResourceVersion
+    {
+        private static bool versionIsLong = true;
+
+        public static int GetVersion(SqlDataReader reader, int index)
+        {
+            int version = 0;
+            retryVersion:
+            try
+            {
+                version = versionIsLong ? (int)reader.GetInt64(index) : reader.GetInt32(index);
+            }
+            catch (InvalidCastException)
+            {
+                versionIsLong = !versionIsLong;
+                goto retryVersion;
```

Comment:
We should probably avoid using a `goto` here, or pretty much anywhere, ever. There is pretty strong industry consensus that if using a goto there is likely a better/simpler solution.

Reply:
Removed goto